google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketContentEditorPanel.java

```diff
@@ -169,18 +171,34 @@ private void showBlobTable() {
    */
   private void loadBlobsStartingWith(String prefix, Consumer<List<Blob>> afterLoad) {
     showLoader();
+    hideError();
     ThreadUtil.getInstance()
         .executeInBackground(
             () -> {
-              List<Blob> blobs =
-                  Lists.newArrayList(
-                      bucket
-                          .list(BlobListOption.currentDirectory(), BlobListOption.prefix(prefix))
-                          .iterateAll());
-              ApplicationManager.getApplication().invokeAndWait(() -> {
-                hideLoader();
-                afterLoad.accept(blobs);
-              });
+              List<Blob> blobs;
+              try {
+                blobs =
+                    Lists.newArrayList(
+                        bucket
+                            .list(BlobListOption.currentDirectory(), BlobListOption.prefix(prefix))
+                            .iterateAll());
+              } catch (StorageException se) {
+                ApplicationManager.getApplication()
+                    .invokeAndWait(
+                        () -> {
+                          hideLoader();
+                          showError();
+                        });
+                return;
+              }
+
+              final List<Blob> loadedBlobs = blobs;
```

Comment:
I'm guessing you have to do this because the compiler can't tell that `blobs` is "effectively final" since its instantiation is separated from its definition?

Reply:
yes. and I can't make blobs final for obvious reasons